source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+      const double depth                       = position.norm() - radius_WGS84(lat_long.first);
+      const double isotherm_depth              = get_isotherm_depth(lat_long.first, lat_long.second);
+      if (depth < isotherm_depth)
+        return isotherm_temperature - (depth - isotherm_depth) * 0.0005;
+      else
+        return surface_temperature + ( depth/isotherm_depth)*(isotherm_temperature - surface_temperature);
+    }
+
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial conditions");
+      {
+        prm.enter_subsection("Adiabatic boundary");
+        {
+          prm.declare_entry ("Isotherm depth filename",
+                             "adiabatic_boundary.txt",
+                             Patterns::FileName(),
+                             "File from which the base of the lithosphere depth data is read. "
+                             "Note that full path (/$DIRECTORY_PATH/filename) of the location "
+                             "of the file must be provided. ");
+          prm.declare_entry ("Isotherm temperature", "1673.15",
+                             Patterns::Double (0),
+                             "The value of the isothermal boundary temperature. Units: Kelvin. ");
+          prm.declare_entry ("Surface temperature", "273.15",
+                             Patterns::Double (0),
+                             "The value of the suface temperature. Units: Kelvin. ");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial conditions");
+      {
+        prm.enter_subsection("Adiabatic boundary");
+        {
+          isotherm_file_name = prm.get("Isotherm depth filename");
+          isotherm_temperature = prm.get_double("Isotherm temperature");
+          surface_temperature  = prm.get_double("Surface temperature");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      std::ifstream input1(isotherm_file_name.c_str());
+      AssertThrow (input1.is_open(),
+                   ExcMessage (std::string("Can't read from file <") + isotherm_file_name + ">"));
+
+      /**
+       * Reading the data file.
+       */
+      while (true)
+        {
+          double latitude_iso, longitude_iso, depth_iso;
+          input1 >> latitude_iso >>longitude_iso >> depth_iso;
+          if (input1.eof())
+            break;
+          latitudes_iso.push_back(latitude_iso);
+          longitudes_iso.push_back(longitude_iso);
+          depths_iso.push_back(depth_iso);
+        }
+      /**
+       * Find first 2 numbers that are different to use in
+       * calculating half the difference between each position as delta.
+       */
+
+      if (std::fabs(latitudes_iso[0] - latitudes_iso[1]) > 1e-9)
```

Comment:
My mistake, I did not see the difference latitude vs longitude in the two branches. Having the same comment in the two branches seemed to me like doing the same thing. Maybe you could specify the comments to "half the latitude distance" and "half the longitude distance" or something similar?